### PR TITLE
GitHub workflows

### PR DESCRIPTION
**Pull request by filipehb**
## Implements

- New workflow to automatic publish new versions in Pypi
- New workflow to run tests on new PR

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Publish Python Package
+
+on:
+  push:
```
Comment (sadikkuzu) [inline .github/workflows/publish.yml:4]: I think it is more frequent that every master commits will have python package publish action. We may consider [on-release-published](https://github.com/sadikkuzu/squiral/blob/main/.github/workflows/python-publish.yml#L6-L8). So we can control pypi publish times via GitHub repo package [releases](https://github.com/serhatbolsu/robotframework-appiumlibrary/releases).

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+jobs:
+  deploy:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python
+      uses: actions/setup-python@v3
```
Comment (sadikkuzu) [inline .github/workflows/publish.yml:16]:
```suggestion
      uses: actions/setup-python@v4
```

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: Publish Python Package
+
+on:
+  push:
+    branches: 
+      - master
```
Comment (sadikkuzu) [inline .github/workflows/publish.yml:6]:
```suggestion
on:
  release:
    types: [published]
```

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install build
+    - name: Build package
+      run: python -m build
+    - name: Publish package
+      uses: pypa/gh-action-pypi-publish@27b31702a0e7fc50959f5ad993c78deac1bdfc29
```
Comment (sadikkuzu) [inline .github/workflows/publish.yml:26]:
```suggestion
      uses: pypa/gh-action-pypi-publish@v1
```

as indicated in [usage](https://github.com/pypa/gh-action-pypi-publish#usage) section.

**Review by sadikkuzu — CHANGES_REQUESTED** [review]
I have some suggestions.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+      
+    - name: Install dependencies
+      run: make
+
```
Comment (sadikkuzu) [inline .github/workflows/ci.yml:17]: Since `make test` installs dependencies, no need to have these lines. Only checkout & run tests steps are enough.

**Comment by sadikkuzu** [comment]
Hi @filipehb 
Can you merge latest `serhatbolsu:master` branch into your `filipehb:github-workflows` branch?
So, Makefile will be fixed and your ci workflow will work fine.

**Comment by serhatbolsu** [comment]
@filipehb <-

File: .github/workflows/publish.yml (hunk 1 above)
Comment (filipehb) [reply .github/workflows/publish.yml]: Good sugestion, eu fiz essa sugestão no último commit

File: .github/workflows/ci.yml (hunk 5 above)
Comment (filipehb) [reply .github/workflows/ci.yml]: done

**Review by sadikkuzu — APPROVED** [review]
Very well done, thanks!
LGTM

**Comment by serhatbolsu** [comment]
remove the `CI` workflow if you like, there is little point of having it while tests are broken

**Comment by serhatbolsu** [comment]
@filipehb  thank you

**Pull request merged**